ci/azure/linux_script

```diff
@@ -59,6 +59,10 @@ unset CXX
 
 make $JOBS install
 
+# look for formatting errors
+release/bin/zig fmt --check ../lib/std ../doc ../src ../tools ../test ../build.zig
+release/bin/zig ast-check ../src/main.zig
```

Comment:
I suggest we only do the zig fmt check for now, for a few reasons:
 * This doesn't actually check any files other than src/main.zig
 * std lib AST checking is already happening with every stage2 test
 * It's an unrelated change to adding zig fmt checking 
 * we may want to consider taking advantage of the thread pool and cache, which is already happening with std lib AST checking

Reply:
>     * This doesn't actually check any files other than src/main.zig

Solved with `find . -name  "*.zig" ! -path "*/zig-cache/*" ! -path "*/zig-out/*" | xargs -n1 release/bin/zig ast-check`.

>     * std lib AST checking is already happening with every stage2 test

Std files can be easily excluded in the command.

>     * It's an unrelated change to adding zig fmt checking

Sure, it did find a some bugs in stage2 and a lot of nonconforming code everywhere else so it might be better to do separately.  

>     * we may want to consider taking advantage of the thread pool and cache, which is already happening with std lib AST checking

It takes less than 15 seconds to check all ~900 files on my laptop on a debug build, I don't think performance is going to be that big of an issue.